styleguide/javascript/README.md

```diff
@@ -0,0 +1,140 @@
+# JavaScript Styleguide
+
+## Tooling
+
+#### Avoid jQuery
+
+Just don't use it. It's better to use standard APIs.
+
+[Should you use or learn jQuery in 2020?](https://flaviocopes.com/jquery/)
+
+#### Avoid lodash
+
+You probably don't need it. And if you do, import the functions you need like this:
+
+``` javascript
+import foo from 'lodash/foo'
+```
+
+Don't do this:
+
+``` javascript
+import { foo } from 'lodash'
+import _ from 'lodash'
+```
+
+The latter will make tree-shaking impossible.
+Read more about [lodash and tree-shaking](https://www.azavea.com/blog/2019/03/07/lessons-on-tree-shaking-lodash/)
+
+#### Use Prettier
+
+Use our [prettier-config](https://github.com/barsoom/prettier-config)
+
+#### Use eslint
+
+Use [eslint-config-barsoom](https://github.com/barsoom/eslint-config-barsoom)
+
+#### When using Babel
+
+Try to avoid using too many experimental features.
+
+Stage 3 is fine and stage 2 can be ok.
+
+Just don't go crazy with everything shiny and new that is likely to change, or maybe not even get into the standard.
+
+#### Prefer npm over yarn
+
+When Yarn was initially released, it solved a lot of problems that npm had.
+
+Since then, npm has improved a lot and nowadays there isn't really any reason to use yarn over npm.
+
+[Why Are You Still Using Yarn in 2018?](https://iamturns.com/yarn-vs-npm-2018/)
```

Comment:
One feature in `yarn` that I use is `resolutions`, and I don't know of a way to control that in `npm`.

Reply:
Never used it myself!

I guess if we have a situation where we need to use that feature we could switch to yarn for that project. After all, this is just a guideline. But I feel like it's better to use npm which comes with most nodejs-installs instead of using a third-party package manager that does 99% of the same things.

Reply:
Looks like a similar feature is coming to npm pretty soon:

https://github.com/npm/rfcs/blob/latest/accepted/0009-package-overrides.md
https://github.com/npm/rfcs/pull/129

Reply:
https://stackoverflow.com/questions/52416312/npm-equivalent-of-yarn-resolutions